src/language_server_protocol.rs

```diff
@@ -2675,13 +2676,26 @@ impl LanguageClient {
                     self.update(|state| {
                         if let Some(ref mut watcher) = state.watchers.get_mut(languageId) {
                             for w in &opt.watchers {
-                                let recursive_mode = if w.glob_pattern.ends_with("**") {
-                                    notify::RecursiveMode::Recursive
-                                } else {
-                                    notify::RecursiveMode::NonRecursive
-                                };
-                                watcher
-                                    .watch(w.glob_pattern.trim_end_matches("**"), recursive_mode)?;
+                                info!("Watching glob pattern: {}", &w.glob_pattern);
+                                for entry in glob(&w.glob_pattern)? {
+                                    match entry {
+                                        Ok(path) => {
+                                            let mode = match path.is_dir() {
+                                                true => notify::RecursiveMode::Recursive,
+                                                false => notify::RecursiveMode::NonRecursive,
+                                            };
```

Comment:
Clippy will complain about this, can we change it to:

```
let mode = if path.is_dir() {
    notify::RecursiveMode::Recursive
} else {
    notify::RecursiveMode::NonRecursive
};
```

intead?

Reply:
Just pushed an update to change it to use `if/else`.